src/config.c

```diff
@@ -76,6 +76,26 @@ struct ifi *config_find_iface(int ifindex)
     return NULL;
 }
 
+static int getmac(const char *ifname, uint8_t *mac, size_t size)
+{
+    struct ifreq ifr;
+    int sock;
+
+    if ((sock = socket(AF_INET, SOCK_DGRAM, 0)) < 0)
+        return 1;
+
+    ifr.ifr_addr.sa_family = AF_INET;
+    strncpy(ifr.ifr_name, ifname, IFNAMSIZ-1);
```

Comment:
Weirdly enough, you don't need to take `IFNAMSIZ-1` into consideration. The max length, including terminating zero, is 16 bytes (`IFNAMSIZ`).  So you can safely use `IFNAMSIZ`.

Also, you should use `strlcpy()` instead if you want to ensure `ifr_name` is zero terminated.  It's available in querierd as a "replacement" function, courtesy of autoconf et al :-)

On the intended target system this may not be an actual issue, but systems using "predictable interface names", i.e. systemd, very often use up MAX characters for interface names.  Come to think of it, maybe the VRRP interface names on the target system also do this ...